examples/src/main/java/io/grpc/examples/helloworld/HelloWorldClient.java

```diff
@@ -0,0 +1,268 @@
+ */
+
+package io.grpc.examples.helloworld;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.util.RoundRobinLoadBalancerFactory;
+import io.grpc.StatusRuntimeException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import java.net.URI;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import javax.annotation.Nullable;
+
+import io.grpc.Attributes;
+import io.grpc.NameResolver;
+import io.grpc.NameResolverProvider;
+import io.grpc.EquivalentAddressGroup;
+//
+// import zookeeper classes
+import java.util.concurrent.CountDownLatch;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.Watcher.Event.KeeperState;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.AsyncCallback.StatCallback;
+import org.apache.zookeeper.KeeperException.Code;
+import org.apache.zookeeper.data.Stat;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.Stat;
+
+
+class ZkNameResolver extends NameResolver implements Watcher {
+  /* Hard-coded path to the ZkNode that knows about servers */
+  public static final String path = "/grpc_hello_world_service";
+  /* 2 seconds to detect that server disconnected */
+  public static final int kTimeoutMs = 2000;
+
+  private static URI zk_uri;
+  private static ZooKeeper zoo;
+  private static Listener listener;
+  private static final Logger logger = Logger.getLogger("ZK");
+
+  /*
+   * The callback from Zookeeper when servers are added/removed.
+   */
+  public void process(WatchedEvent we) {
+    if (we.getType() == Event.EventType.None) {
+      logger.info("Connection expired");
+    } else {
+      try {
+        List<String> servers  = zoo.getChildren(path, false);
+        AddServersToListener(servers);
+        zoo.getChildren(path, this);
+      } catch(Exception ex) {
+        logger.info(ex.getMessage());
+      }
+    }
+  }
+
+  public void AddServersToListener(List<String> servers) {
+    List<EquivalentAddressGroup> addrs = new ArrayList<EquivalentAddressGroup>();
+    logger.info("Updating server list");
+    for (String child : servers) {
+      try {
+        logger.info("Online: " + child);
+        URI uri = new URI("dummy://" + child);
+        /* Convert "host:port" into host and port */
+        String host = uri.getHost();
+        int port = uri.getPort();
+        List<SocketAddress> sockaddrs_list= new ArrayList<SocketAddress>();
+        sockaddrs_list.add(new InetSocketAddress(host, port));
+        EquivalentAddressGroup eag =  new EquivalentAddressGroup(sockaddrs_list);
+        addrs.add(eag);
+      } catch(Exception ex) {
+        logger.info("Unparsable server address: "+child);
```

Comment:
It would work but you'd have to create regex for IPV4, IPV6 and any other address schemes separately. this seemed like a good way.